include/measurement_kit/common/libevent/socket.hpp

```diff
@@ -0,0 +1,377 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software under the BSD license. See AUTHORS
+// and LICENSE for more information on the copying conditions.
+#ifndef MEASUREMENT_KIT_COMMON_LIBEVENT_SOCKET_HPP
+#define MEASUREMENT_KIT_COMMON_LIBEVENT_SOCKET_HPP
+
+#include <cassert>
+#include <deque>
+#include <event2/buffer.h>
+#include <event2/bufferevent.h>
+#include <event2/event.h>
+#include <event2/util.h>
+#include <measurement_kit/common/detail/mock.hpp>
+#include <measurement_kit/common/detail/utils.hpp>
+#include <measurement_kit/common/reactor.hpp>
+#include <measurement_kit/common/socket.hpp>
+#include <mutex>
+
+extern "C" {
+static inline void mk_socket_delete_string(const void *, size_t, void *);
+static inline void mk_socket_connect(bufferevent *, short, void *);
+static inline void mk_socket_event(bufferevent *, short, void *);
+static inline void mk_socket_flush(bufferevent *, void *);
+static inline void mk_socket_read(bufferevent *, void *);
+}
+
+namespace mk {
+inline namespace libevent {
+
+template <MK_MOCK(evbuffer_new), MK_MOCK(evbuffer_add_reference),
+        MK_MOCK(evbuffer_add_buffer), MK_MOCK(evbuffer_pullup),
+        MK_MOCK(evbuffer_get_length)>
+class LibeventSocketBuffer : public mk::SocketBuffer {
```

Comment:
This is basically a simplified `mk::net::Buffer`. We should use `mk::net::Buffer` instead.